pkg/cluster/deploybaseresources.go

```diff
@@ -178,6 +185,54 @@ func (m *manager) deployBaseResourceTemplate(ctx context.Context) error {
 	return arm.DeployTemplate(ctx, m.log, m.deployments, resourceGroup, "storage", t, nil)
 }
 
+// subnetsWithServiceEndpoint returns a unique slice of subnet resource IDs that have the corresponding
+// service endpoint
+func (m *manager) subnetsWithServiceEndpoint(ctx context.Context, serviceEndpoint string) ([]string, error) {
+	subnetsMap := map[string]struct{}{}
+
+	subnetsMap[m.doc.OpenShiftCluster.Properties.MasterProfile.SubnetID] = struct{}{}
+	workerProfiles, _ := api.GetEnrichedWorkerProfiles(m.doc.OpenShiftCluster.Properties)
+	for _, v := range workerProfiles {
+		// don't fail empty worker profiles/subnet IDs as they're not valid
+		if v.SubnetID == "" {
+			continue
+		}
+
+		subnetsMap[strings.ToLower(v.SubnetID)] = struct{}{}
+	}
+
+	// TODO: service endpoints can be region specific.  Will this cause a problem
```

Comment:
I think so - the wording [here](https://learn.microsoft.com/en-us/azure/storage/common/storage-network-security?toc=%2Fazure%2Fvirtual-network%2Ftoc.json&tabs=azure-portal#grant-access-from-a-virtual-network) suggests that when connecting a vnet in region A to a storage account in region B, you need a cross-region service endpoint.

GIven that these just went GA a few months back, I am assuming we are using service endpoints classic which are region specific.

Reply:
I accounted for both - need to remove the comments.

Reply:
For my own knowledge then - in [this loop](https://github.com/Azure/ARO-RP/pull/3211/files#diff-757380ec654f144c2f36b9c7f8c8ce2ca911fb9d054d258aa81c447c47ff7de6R222) we append location if resource type is `Microsoft.Storage`, but just skip updating it if resource type is `Microsoft.Storage.Global` since it doesn't need a location - and this handles both service endpoint types. Is that roughly correct?